src/core/a-entity.js

```diff
@@ -704,29 +690,33 @@ var proto = Object.create(ANode.prototype, {
                              HTMLElement.prototype.getAttribute.call(this, attrName));
       }
 
-      // Determine which type of setAttribute to call based on the types of the arguments.
-      if (arg1Type === 'string' && typeof arg2 !== 'undefined') {
-        singlePropertyUpdate(this, attrName, arg1, arg2);
+      // Determine new attributes from the arguments
+      if (
+        typeof arg2 !== 'undefined' &&
+        typeof arg1 === 'string' &&
+        arg1.length > 0 &&
+        (typeof utils.styleParser.parse(arg1)) === 'string'
```

Comment:
Thoughts for alternatives here? Should the work be done to `component.updateProperties` to allow this parsed value to be passed directly (and if so, would it make more sense to go ahead and use `component.parse`)?